### PR TITLE
Support firmware update

### PR DESCRIPTION
**Pull request by aounhaider1**
fixes https://github.com/eclipse/smarthome/issues/2606
Signed-off-by: Aoun Bukhari <bukhari@itemis.de>

### CODE REVIEW

**Comment by shoeper** [comment]
Some documentation could help other devs to understand the coder better & faster.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareDTO.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.eclipse.smarthome.core.thing.firmware;
+
```
Comment (htreu) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareDTO.java]: License Header & JavaDoc missing

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareDTO.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public String firmwareUID;
+    public String vendor;
+    public String model;
+    public String description;
+    public String version;
+    public String changelog;
+    public String prerequisiteVersion;
+
+    public FirmwareDTO() {
+
```
Comment (htreu) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareDTO.java]: remove empty line

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareStatusDTO.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package org.eclipse.smarthome.core.thing.firmware;
+
```
Comment (htreu) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareStatusDTO.java]: - Lisence Header mssing
- move file to core.thing.firmware.dto package

**Comment by kaikreuzer** [comment]
Superseded by https://github.com/eclipse/smarthome/pull/3206.

**Pull request closed without merging**